google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManagementConfirmationDialog.java

```diff
@@ -127,12 +136,38 @@
     roleTable.setTableHeader(null);
     rolePane.setBorder(JBUI.Borders.empty());
     rolePanel.setVisible(!roles.isEmpty());
+
+    serviceAccountNameTextField.setText(module.getName());
+
+    serviceKeyPathSelector.addBrowseFolderListener(
+        GctBundle.message("cloud.apis.management.dialog.serviceaccount.key.browser.title"),
+        null /*description*/,
+        null /*project - null on purpose so that the path isn't defaulted to the project root*/,
+        FileChooserDescriptorFactory.createSingleFolderDescriptor());
+    serviceKeyPathSelector.setText(getServiceKeyPathSelectorDefaultText());
   }
 
   Set<Role> getSelectedRoles() {
     return ((ServiceAccountRolesTableModel) roleTable.getModel()).getSelectedRoles();
   }
 
+  @Nullable
+  @Override
+  protected ValidationInfo doValidate() {
+    if (newServiceAccountCheckbox.isSelected()) {
+      if (StringUtils.isEmpty(serviceAccountNameTextField.getText())) {
+        return new ValidationInfo(
+            GctBundle.message("cloud.apis.management.dialog.serviceaccount.name.error"),
+            serviceAccountNameTextField);
+      } else if (StringUtils.isEmpty(serviceKeyPathSelector.getText())) {
```

Comment:
What happens if the service key path is not valid? Like when a user types in an invalid path

Reply:
yeah good point I should probably check to make sure its a valid directory too

Reply:
done: e1f86da